src/server/game/Entities/Player/Player.cpp

```diff
@@ -13396,7 +13399,12 @@ void Player::RemoveItemFromBuyBackSlot(uint32 slot, bool del)
         {
             pItem->RemoveFromWorld();
             if (del)
+            {
                 pItem->SetState(ITEM_REMOVED, this);
+                if (ItemTemplate const* itemTemplate = pItem->GetTemplate())
```

Comment:
This line has a crash since using `SetState` above will delete the `pItem`
https://github.com/TrinityCore/TrinityCore/blob/9dc3de10f0044c35a95e9b72e7b874f4b5b8e867/src/server/game/Entities/Item/Item.cpp#L674

Must move the SetState below the rest of the IF statements.